### DISCUSSION TITLE
File picker: ignore git submodules

### DISCUSSION THREAD

**Discussion by beeb** [helix-editor/helix#9078] [General] [answered]
The [file picker section](https://docs.helix-editor.com/configuration.html?highlight=file-picker#editorfile-picker-section) of the config provides much-useful options to disable showing certain types of hidden or git files in the file picker.

However, I would love to have an option to disable all files located inside any git submodule. Take for instance [foundry](https://github.com/foundry-rs/foundry), which uses git submodules for project libraries/dependencies. When I add e.g. `forge-std` as a git submodule, which gets cloned into the `lib` folder, then all the source files for `forge-std` show up in my file picker. 

I couldn't find an easy way to hide those, as obviously I can't git-ignore a git submodule.

Thanks!

**Comment by kirawi** [comment, marked as answer]
You can use a `.helix/ignore` file in 23.10+

**Reply by beeb** [reply]
Thanks! Not ideal, as I probably then need to either add this ignore file to .gitignore or publish it to the repo, but it will do.

**Reply by beeb** [reply]
in fact I will add .helix/ignore to my global gitignore so it's fine!

**Reply by novusnota** [reply]
@beeb you can create and use a global `~/.config/helix/ignore` file in your Helix config directory as well, see: #9102

**Reply by beeb** [reply]
Thanks for the hint!

**Comment by pascalkuthe** [comment]
Yeah custom ignore logic like that is hard to add because it's not supported by the ignore crate. I also think while this may be useful in this one generic instance it doesn't seem generally useful (or a common ignore convention). Using .helix/ignore seems more appropriate

**Comment by willbush** [comment]
I randomly wanted to point out helix picker seems respect `.ignore` files as well as other tools like ripgrep. nvim fzf-lua picker seems to also (or perhaps because I'm using sk instead of fzf). Found this a nice alternative to editor specific solution. Using `.gitignore` feels wrong since [submodule ignore ](https://git-scm.com/docs/gitmodules#Documentation/gitmodules.txt-submodulenameignore) exists, but I'm not sure it would be a problem in practice.
